src/boot/efi/efi-string.c

```diff
@@ -272,6 +274,453 @@ bool efi_fnmatch(const char16_t *pattern, const char16_t *haystack) {
 DEFINE_PARSE_NUMBER(char, parse_number8);
 DEFINE_PARSE_NUMBER(char16_t, parse_number16);
 
+enum {
+        SCRATCH_BUF_SIZE = 64,
```

Comment:
can we determine how much precisely we need at max here? i.e. largest 64bit number in octal with most decoration we can have?